std/socket.d

```diff
@@ -1269,6 +1269,14 @@ abstract class Address
     /// Returns actual size of underlying $(D sockaddr) structure.
     abstract @property socklen_t nameLen() const pure nothrow @nogc;
 
+    // Sets the actual size of underlying $(D sockaddr).
+    // Kludge for setting path length in UnixAddress
+    protected @property void nameLen(socklen_t len) pure
+    {
+        if (len > this.nameLen)
+            throw new SocketParameterException("Not enough socket address storage");
+    }
+
```

Comment:
Alright. So, abstract UNIX socket addresses require that the address length is not greater than the length of the actual abstract "path", because they allow embedded null bytes and there is no way to infer the address length like the other address types we support so far. For this, I think a regular method would be more appropriate (for the reasons I mentioned above).

Also, it would be great if you could summarize the above into comments, as all this may not be obvious to the next person looking at this code.